### PR TITLE
Add a feature and refactor

### PR DESCRIPTION
**Pull request by amirmalekian**
This pull request includes changes to the Sudoer_VPN_bot. I've added a feature to show a VPN zone name with a VPN URL when the user is entering the 'وضعیت من چیه؟'.

I've also refactored and prevented duplicate code which reduces about 30 lines of code.

Feel free to let me know if you need any more information or if there's anything else I can do to help get this merge.

### CODE REVIEW

File: insert_urls.py [hunk 1]
```diff
@@ -6,14 +6,12 @@
...
-    conn = None
     try:
-        conn = sqlite3.connect(db_file)
-        return conn
+        return sqlite3.connect(db_file)
     except Error as e:
         print(e)
 
-    return conn
+    return None
```
Comment (hasanisaeed) [inline insert_urls.py:14]: By default, it returns a `None` value.  No need for `return None`

**Comment by MortezaBashsiz** [comment]
Thanks

I have tested but in 
وضعیت من چیه؟
it does not mention the name of zone 
Have you tested it?

one another note it could be great if you choose ps instead of hostname as zone name. fot this you can simply decode the url after "vmess://" by base64 and with choose field ps from json

File: insert_urls.py (hunk 1 above)
Comment (amirmalekian) [reply insert_urls.py:14]: no no :)
By default, None is not be returned
If it goes to the except block, it only prints the error
But in the main function, we consider this function as the value of the con variable.

On the other hand, if you see the PR code, I deleted the conn variable from inside the create_connection function
But in the repo code, the con variable is returned, which is the default value of None.

**Comment by amirmalekian** [comment]
you're welcome :)

It is a good suggestion, I agree.
I will do it.

**Comment by amirmalekian** [comment]
I implemented URL decoding by base64. 
Please check. @MortezaBashsiz

**Comment by MortezaBashsiz** [comment]
It faced with issue
Please test it
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/dist-packages/python_telegram_bot-20.0a6-py3.9.egg/telegram/ext/_application.py", line 1057, in process_update
    await coroutine
  File "/usr/local/lib/python3.9/dist-packages/python_telegram_bot-20.0a6-py3.9.egg/telegram/ext/_conversationhandler.py", line 824, in handle_update
    new_state: object = await handler.handle_update(
  File "/usr/local/lib/python3.9/dist-packages/python_telegram_bot-20.0a6-py3.9.egg/telegram/ext/_handler.py", line 143, in handle_update
    return await self.callback(update, context)
  File "/tmp/Sudoer_VPN_bot/bot.py", line 417, in status
    for url in urls:
TypeError: 'int' object is not iterable

```

**Comment by amirmalekian** [comment]
I think you run another file.
Because the for loop index is a `row` instead of `url`.

**Comment by MortezaBashsiz** [comment]
> I think you run another file. Because the for loop index is a `row` instead of `url`.

Yes I was in wrong branch
But again with your develop branch failed

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/dist-packages/python_telegram_bot-20.0a6-py3.9.egg/telegram/ext/_application.py", line 1057, in process_update
    await coroutine
  File "/usr/local/lib/python3.9/dist-packages/python_telegram_bot-20.0a6-py3.9.egg/telegram/ext/_conversationhandler.py", line 824, in handle_update
    new_state: object = await handler.handle_update(
  File "/usr/local/lib/python3.9/dist-packages/python_telegram_bot-20.0a6-py3.9.egg/telegram/ext/_handler.py", line 143, in handle_update
    return await self.callback(update, context)
  File "/tmp/Sudoer_VPN_bot/bot.py", line 386, in status
    for row in urls:
TypeError: 'int' object is not iterable

```

**Comment by MortezaBashsiz** [comment]
Again issue when I get a new vpn

```py
2023-01-31 07:13:44,241 - telegram.ext._application - ERROR - No error handlers are registered, logging exception.
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/dist-packages/python_telegram_bot-20.0a6-py3.9.egg/telegram/ext/_application.py", line 1057, in process_update
    await coroutine
  File "/usr/local/lib/python3.9/dist-packages/python_telegram_bot-20.0a6-py3.9.egg/telegram/ext/_conversationhandler.py", line 824, in handle_update
    new_state: object = await handler.handle_update(
  File "/usr/local/lib/python3.9/dist-packages/python_telegram_bot-20.0a6-py3.9.egg/telegram/ext/_handler.py", line 143, in handle_update
    return await self.callback(update, context)
  File "/tmp/Sudoer_VPN_bot/bot.py", line 275, in get_config
    zone_reg, url = get_url_byzone(conn, region)
TypeError: cannot unpack non-iterable int object


```

**Comment by MortezaBashsiz** [comment]
Please
Test it
I can not do it again
```py
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/dist-packages/python_telegram_bot-20.0a6-py3.9.egg/telegram/ext/_application.py", line 1057, in process_update
    await coroutine
  File "/usr/local/lib/python3.9/dist-packages/python_telegram_bot-20.0a6-py3.9.egg/telegram/ext/_conversationhandler.py", line 824, in handle_update
    new_state: object = await handler.handle_update(
  File "/usr/local/lib/python3.9/dist-packages/python_telegram_bot-20.0a6-py3.9.egg/telegram/ext/_handler.py", line 143, in handle_update
    return await self.callback(update, context)
  File "/tmp/Sudoer_VPN_bot/bot.py", line 296, in get_config
    inc_url_used_count(conn, url)
  File "/tmp/Sudoer_VPN_bot/bot.py", line 91, in inc_url_used_count
    cur.execute(sql, url)
ValueError: parameters are of unsupported type
```

**Comment by amirmalekian** [comment]
There is a problem with running the code, which is why I can't take the test.

**Comment by MortezaBashsiz** [comment]
Works fine
But there is a problem
When it returns the url, the message contains the name also
like as following

image

It could be great if you put it in `URL` to be copied with one touch
otherwise the user must select only vmes:// url and this will make trouble for user

**Comment by amirmalekian** [comment]
Regarding the problem you mentioned, after decoding some URLs, the value of the `ps` field is set to the same `Sudoer_VPN_bot `as the following URL:

`vmess://ewoiYWRkIjoiY2Yuc3Vkb2VyLm5ldCIsIAoiYWlkIjoiMCIsIAoiYWxwbiI6IiIsIAoiaG9zdCI6InNjaGVyZWh0emZsazAxLnNjaGVyZS5uZXQiLCAKImlkIjoiYWUxOGQ2YzgtOWU5Ni01MmYxLWIxY2ItMDZkZDgyYTZmNzUxIiwgCiJuZXQiOiJ3cyIsIAoicGF0aCI6Ii9hcGkwNiIsIAoicG9ydCI6IjQ0MyIsIAoicHMiOiJTdWRvZXJfVlBOX2JvdCIsIAoic2N5IjoiYXV0byIsIAoic25pIjoiMzgzZDE2NzYtOWZiZi0xMWVkLThkMjQtMmZkYTBkNTRhMDFlLnNjaGVyZS5uZXQiLCAKInRscyI6InRscyIsIAoidHlwZSI6IiIsIAoidiI6IjIiIAp9Cg==`

Decoded result:

```
{
'add': 'cf.sudoer.net',
 'aid': '0',
 'alpn': '',
 'host': 'scherehtzflk01.schere.net',
 'id': 'ae18d6c8-9e96-52f1-b1cb-06dd82a6f751',
 'net': 'ws',
 'path': '/api06',
 'port': '443',
 'ps': 'Sudoer_VPN_bot',
 'scy': 'auto',
 'sni': '383d1676-9fbf-11ed-8d24-2fda0d54a01e.schere.net',
 'tls': 'tls',
 'type': '',
 'v': '2'
 }
```
And this is no longer in my hands :)
This is the field you are looking for.

**Comment by MortezaBashsiz** [comment]
Yes
I mean provide the url in back-coute or something else to be able to copy it with one single touch on the message

**Comment by amirmalekian** [comment]
I don't know if the `python-telegram-bot` library has this feature or not, but you can use other packages like `pyperclip` or even handle this by adding a button like the one below:


```
async def get_config(update: Update, context: ContextTypes.DEFAULT_TYPE) -> None:
    ...
    if user_url == 0:
        zone_reg, url = get_url_byzone(conn, region)
        if url is None:
            result = "متاسفانه این منطقه ظرفیتش تکمیل شده لطفا جاهای دیگه رو امتحان کن"
            await update.callback_query.message.reply_text(result)
        else:
            inc_url_used_count(conn, url)
            insert_user_url(conn, user.id, url[0])
            result = """
            مقدار url پایین رو کپی کنید و در برنامه اضافه بکنید
            """
            url_button = InlineKeyboardButton("Copy URL", callback_data=url[0])
            keyboard = [[url_button]]
            reply_markup = InlineKeyboardMarkup(keyboard)
            result = f"{zone_reg}:\n{url[0]}"
            await update.callback_query.message.reply_text(result, reply_markup=reply_markup)
    else:
        result = "تو قبلا از این منطقه فیلترشکن گرفتی برای اینکه ببینیش از منوی اصلی گزینه (وضعیت من چیه؟) رو انتخاب کن"
        await update.callback_query.message.reply_text(result)
```

Since the code doesn't run for me and I can't see the output of the code, we may encounter many ridiculous bugs again (:
That's why I'm excused from doing it.

**Pull request merged**